### ISSUE TITLE
IDE Fatal Error while indexing

### ISSUE THREAD

**Issue by nntoan** [magento/magento2-phpstorm-plugin#261]
**Describe the bug** (*) 

IDE throws error while indexing.

**To Reproduce** (*) 

While indexing is running, it throws error in Event Log

```
Can't find template Magento CLI Command

com.intellij.diagnostic.PluginException: Template not found: Magento CLI Command [Plugin: com.magento.idea.magento2plugin]
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getInternalTemplates(FileTemplateManagerImpl.java:222)
	at com.intellij.ide.fileTemplates.impl.AllFileTemplatesConfigurable.<init>(AllFileTemplatesConfigurable.java:96)
	at com.intellij.ide.fileTemplates.impl.AllFileTemplatesConfigurable$Provider.createConfigurable(AllFileTemplatesConfigurable.java:62)
	at com.intellij.openapi.options.ConfigurableEP$ProviderProducer.createElement(ConfigurableEP.java:379)
	at com.intellij.openapi.options.ConfigurableEP.createConfigurable(ConfigurableEP.java:309)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.createConfigurable(ConfigurableWrapper.java:47)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.getConfigurable(ConfigurableWrapper.java:109)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.cast(ConfigurableWrapper.java:84)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.prepare(ConfigurableCardPanel.java:38)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.prepare(ConfigurableCardPanel.java:29)
	at com.intellij.ui.CardLayoutPanel.lambda$selectLater$1(CardLayoutPanel.java:114)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:255)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Template not found: Magento CLI Command
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getTemplateFromManager(FileTemplateManagerImpl.java:285)
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getJ2eeTemplate(FileTemplateManagerImpl.java:271)
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getInternalTemplate(FileTemplateManagerImpl.java:234)
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getInternalTemplates(FileTemplateManagerImpl.java:219)
	... 17 more
```

**Expected behavior** (*) 

No error.

**Screenshots**

**Please complete the following information:** (*) 

 - OS: MacOS 10.13
 - PhpStorm/Intellij version 2020.1.2
 - Version 1.0.1

**Additional context**

**Comment by coderimus** [comment]
Hello @nntoan ,
Thank you for the report. I checked the 1.0.1 version locally (macOS Catalina v10.15.4, PHPStorm 2020.1.2) and was not able to reproduce the reported issue. I reinstall the plugin, invalidate the cache and restart IDE to run the full project reindex. Before this, I checked that the plugin is turned on and ready to work. I also run reindex in the scope of the module settings.

Can you please check that you use the version installed from the Intellij marketplace, this version is up today?

**Comment by nntoan** [comment]
Hello @coderimus ,

This one might be very hard to reproduce, even for me. I've tried to reindex number of projects and also cannot reproduce this issue. My PhpStorm is installed by Jetbrains Toolbox anyway.

I think it's safe to close this issue and we can reopen it in the future if more people having this same issue.

Cheers.

**Comment by coderimus** [comment]
@nntoan thank you for the test and report. If this issue appears once, please report back to me. Thank you :)

**Comment by artmouse** [comment]
Can't find template Magento CLI Command

com.intellij.diagnostic.PluginException: Template not found: Magento CLI Command [Plugin: com.magento.idea.magento2plugin]
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getInternalTemplates(FileTemplateManagerImpl.java:222)
	at com.intellij.ide.fileTemplates.impl.AllFileTemplatesConfigurable.<init>(AllFileTemplatesConfigurable.java:96)
	at com.intellij.ide.fileTemplates.impl.AllFileTemplatesConfigurable$Provider.createConfigurable(AllFileTemplatesConfigurable.java:62)
	at com.intellij.openapi.options.ConfigurableEP$ProviderProducer.createElement(ConfigurableEP.java:379)
	at com.intellij.openapi.options.ConfigurableEP.createConfigurable(ConfigurableEP.java:309)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.createConfigurable(ConfigurableWrapper.java:47)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.getConfigurable(ConfigurableWrapper.java:109)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.cast(ConfigurableWrapper.java:84)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.prepare(ConfigurableCardPanel.java:38)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.prepare(ConfigurableCardPanel.java:29)
	at com.intellij.ui.CardLayoutPanel.lambda$selectLater$1(CardLayoutPanel.java:114)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:255)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Template not found: Magento CLI Command
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getTemplateFromManager(FileTemplateManagerImpl.java:285)
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getJ2eeTemplate(FileTemplateManagerImpl.java:271)
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getInternalTemplate(FileTemplateManagerImpl.java:234)
	at com.intellij.ide.fileTemplates.impl.FileTemplateManagerImpl.getInternalTemplates(FileTemplateManagerImpl.java:219)
	... 17 more

**Comment by nntoan** [comment]
I'm having this again and also @artmouse. This is happening very randomly, not sure what steps to reproduce.

![Screenshot 2020-07-25 01 58 40](https://user-images.githubusercontent.com/3891429/88425989-8c81a680-ce1a-11ea-9665-a4bdacd93816.png)

**Comment by vass30** [comment]
The same for :
https://user-images.githubusercontent.com/4979637/89859826-96413300-dbaa-11ea-9266-4a623c447e69.png

**Comment by VitaliyBoyko** [comment]
Fixed in #274
Thank you, guys.
I keep you posted in the slack channel regarding the alfa version with the fix.
